sdk/core/System.ClientModel/samples/ServiceMethods.md

```diff
@@ -0,0 +1,109 @@
+# System.ClientModel-based client service methods
+
+## Introduction
+
+`System.ClientModel`-based clients, or **service clients**, provide an interface to cloud services by translating library calls to HTTP requests.
+
+In service clients, there are two ways to expose the schematized body in the request or response, known as the **message body**:
+
+- **Convenience methods** take strongly-typed models as parameters.  These models are C# classes which map to the message body of the REST call.
+
+- **Protocol method** take primitive types as parameters and their `BinaryContent` input parameters mirror the message body directly. Protocol methods provide more direct access to the HTTP API protocol used by the service.
+
+## Convenience methods
+
+**Convenience methods** provide a convenient way to invoke a service operation.  They are service methods that take a strongly-typed model representing schematized data sent to the service as input, and return a strongly-typed model representing the payload from the service response as output. Having strongly-typed models that represent service concepts provides a layer of convenience over working with the raw payload format. This is because these models unify the client user experience when cloud services differ in payload formats.  That is, a client-user can learn the patterns for strongly-typed models that `System.ClientModel`-based clients provide, and use them together without having to reason about whether a cloud service represents resources using, for example, JSON or XML formats.
+
+The following sample illustrates how to call a convenience method and access both the strongly-typed output model and the details of the HTTP response.
+
+```C# Snippet:ClientResultTReadme
+// Create a client
+string key = Environment.GetEnvironmentVariable("MAPS_API_KEY") ?? string.Empty;
```

Comment:
I would drop the part that sets this to empty string. There does not seem to be a good reason to do it.

Reply:
I'm happy to do that.  Question, though.  If the environment variable isn't set, that call returns null.  Should we ignore that for the purposes of the sample?

Reply:
Yeah, I would ignore. We don't handle errors in samples.

Reply:
Addressed